lib/schemaUtils.js

```diff
@@ -3035,7 +3034,7 @@ module.exports = {
         resolvedParamValue,
         index = _.findIndex(determinedPathVariables, pathVar);
 
-      schemaPathVar = _.find(schemaPathVariables, (param) => {
+      const schemaPathVar = _.find(schemaPathVariables, (param) => {
```

Comment:
To look into `closures` for better implementation rather than defining inside loop.

Reply:
As the concept of closure is not needed here because we are not using variable in question outside of `async.map` scope, keeping solution same as before.